File: app/src/main/java/com/mandarin/bcu/androidutil/adapters/MediaPrepare.java
package com.mandarin.bcu.androidutil.adapters;

import android.media.MediaPlayer;

import com.mandarin.bcu.androidutil.battle.sound.SoundPlayer;

public abstract class MediaPrepare implements MediaPlayer.OnPreparedListener {

    public abstract void PrePare(MediaPlayer mp);

    @Override
    public void onPrepared(MediaPlayer mp) {
        ((SoundPlayer)mp).setPrepared(true);

        PrePare(mp);
    }

}

File: app/src/main/java/com/mandarin/bcu/androidutil/adapters/MediaPrepare.kt
package com.mandarin.bcu.androidutil.adapters

import android.media.MediaPlayer
import android.media.MediaPlayer.OnPreparedListener
import com.mandarin.bcu.androidutil.battle.sound.SoundPlayer

abstract class MediaPrepare : OnPreparedListener {
    abstract fun prepare(mp: MediaPlayer?)
    override fun onPrepared(mp: MediaPlayer) {
        (mp as SoundPlayer).setPrepared(true)
        prepare(mp)
    }
}